File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        l=[]
        d={')':'(',']':'[','}':'{'}
        opens='({['
        close='}])'
        for i in s:
            if i in opens:
                l.append(i)
            if i in close:
                if len(l)==0:
                    return False
                else:
                   # print(l[-1],d[i])
                    if d[i]==l[-1]:
                        l.pop()
                    else:
                        return False
        if len(l)==0:
            return True
        else:
            return False

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Map<Character, Character> mp = new HashMap<>();
        mp.put('}','{');
        mp.put(')','(');
        mp.put(']','[');

        String open="{[(";
        String close="}])";
        Stack<Character> st = new Stack<>();

        for(int i=0;i<s.length();i++){
            char ch=s.charAt(i);
            if(open.contains(String.valueOf(ch))){
                st.push(ch);
            }
           if(close.contains(String.valueOf(ch))){
                if(st.isEmpty()){
                    return false;
                }
                else{
                    if (mp.get(ch)==st.peek()){
                        st.pop();
                    }
                    else{
                        return false;
                    }
                }

           }
        }
        if(st.isEmpty()){
            return true;
        }
        return false;
    }
}